src/Tronkell/Server/Server.hs

```diff
@@ -42,8 +42,13 @@ startServer gConfig = do
   -- start network
   wsThread  <- forkIO $ W.start firstUId networkChans clientsChan
   tcpThread <- forkIO $ Tcp.start firstUId networkChans clientsChan
+  -- to avoid space msg leak..
+  forkIO . forever . readChan $ clientSpecificOutChan
+  forkIO . forever . readChan $ clientsChan
+  forkIO . forever . readChan $ internalChan
```

Comment:
I think github ate my reply, the last time. This feels a bit clunky. Do we not have a better way to do broadcast?

Either way, I think this is okay to push and add as a FIXME. Don't want to block the other changes you have.